src/pages/Library.tsx

```diff
@@ -88,6 +151,13 @@ class Library extends React.Component<ILibraryProps, ILibraryState>
     setContext = (ctx: ILibraryContext) =>
     {
         console.log("setting context: ", ctx);
+        console.log(this.state.showArrows)
+        
+        //Reset various
+        this.setState(update(this.state, {
```

Comment:
Is there a reason why the immutability helper is used?
We usually use it when a state field is a deep object, which needs to be deep copied and substituted in.
If the state values are shallow, and in this case primitive, there shouldnt be a need to use it.

I think this.setState({showFab: false}) should work fine.

Reply:
Nope, just wasn't thinking... I think I still have a lot to learn in React tbh.